src/install-opentofu.ps1

```diff
@@ -0,0 +1,502 @@
+<#
+.SYNOPSIS
+Install OpenTofu.
+.DESCRIPTION
+This script installs OpenTofu via any of the supported methods. Please run it with the -h or -help parameter
+to get a detailed help description.
+.LINK
+https://opentofu.org
+.LINK
+https://opentofu.org/docs/intro/install/
+.PARAMETER help
+Show a more detailed help.
+.PARAMETER installMethod
+The installation method to use. Must be one of:
+- standalone
+.PARAMETER installPath
+Installs OpenTofu to the specified path. (Standalone installation only.)
+.PARAMETER opentofuVersion
+Installs the specified OpenTofu version. (Standalone installation only.)
+.PARAMETER cosignPath
+Path to cosign. (Standalone installation only.)
+.PARAMETER cosignOidcIssuer
+OIDC issuer for cosign signatures. (Standalone installation only.)
+.PARAMETER cosignIdentity
+Identity for the cosign signature. (Standalone installation only.)
+.PARAMETER skipVerify
+Skip cosign integrity verification. (Standalone installation only; not recommended.)
+.Parameter skipChangePath
+Skip changing the user/system PATH variable to include OpenTofu.
+.Parameter allUsers
+Install for all users with elevated privileges.
+.Parameter internalContinue
+Internal parameter to use for continuing with elevated privileges. Do not use.
+.Parameter internalZipFile
+Internal parameter to use for continuing with elevated privileges. Do not use.
+.EXAMPLE
+PS> .\install-opentofu.ps1 -installMethod standalone
+#>
+param(
+    [Parameter(Mandatory = $false)]
+    [switch]$help = $false,
+    [Parameter(Mandatory = $false)]
+    [string]$installPath = "",
+    [Parameter(Mandatory = $false)]
+    [string]$opentofuVersion = "latest",
+    [Parameter(Mandatory = $false)]
+    [string]$installMethod,
+    [Parameter(Mandatory = $false)]
+    [string]$cosignPath = "cosign.exe",
```

Comment:
I've done this so the defaults appear in the `Get-Help` output. Unfortunately, some defaults are dynamic (such as the `installPath` default depending on the presence of the `allUsers` flag) and there is no good way to do this. The default values are also needed for the usage documentation.

Reply:
Yeah - setting the defaults here so that they appear in the help output makes sense. I just mean that maybe instead of doing stuff like this later in the script:

```powershell
if (!$cosignPath) {
    $cosignPath = $defaultCosignPath
}
```

You could actually just fail with an error message if no path is specified. That way the default is only set in a single place.

I totally get that you can't do it for stuff like `installPath`.

Reply:
Cosign doesn't have any information on how to install it on Windows, which is a shame. I have it installed in the path, which is why I built it this way. Not sure how others use it, maybe that's something we can improve based on user feedback? We check if this path is valid anyway, if it isn't the user will get an error message indicating that.